Kubernetes/windows/debug/TroubleShootContainerNetworking.ps1

```diff
@@ -0,0 +1,848 @@
+            }
+
+            if ($lbPolicy.ServiceType -eq [ServiceType]::NodePort) {
+                $allowNatRuleName = "ALLOW_NAT_\w*_{0}_{1}_{2}" -f $networkData.ManagementIpAddress, $lbPolicy.ExternalPort, $lbPolicy.Protocol
+            }
+            else {
+                $allowNatRuleName = "ALLOW_NAT_\w*_{0}_{1}_{2}" -f $lbPolicy.VIP, $lbPolicy.ExternalPort, $lbPolicy.Protocol
+            }
+            
+            foreach($endpoint in $endpointsData)
+            {
+                if ($endpoint.IsRemoteEndpoint -ne $true)
+                {
+                    if ($this.IsVfpRuleConfigured($endpoint.VfpPort, $ruleName, "LB_DSR_IPv4_OUT", "LB_DSR") -eq $false)
+                    {
+                        $this.Status = [TestStatus]::Failed
+                        $this.RootCause += ",Vfp port for POD with IP {0} has missing rule {1}" -f $endpoint.IPAddress, $ruleName
+                        $this.Resolution = "Restart HNS service by executing: Restart-Service -f HNS"
+                    }
+                    
+                    if ($lbPolicy.EndpointIpAddresses.Contains($endpoint.IPAddress))
+                    {
+                        if ($this.IsVfpRuleConfigured($endpoint.VfpPort, $natRuleName, "SLB_GROUP_NAT_IPv4_IN", "SLB_NAT_LAYER") -eq $false)
+                        {
+                            $this.Status = [TestStatus]::Failed
+                            $this.RootCause += ",Vfp port for POD with IP {0} has missing rule {1}" -f $endpoint.IPAddress, $ruleName
+                            $this.Resolution = "Restart HNS service by executing: Restart-Service -f HNS"
+                        }
+    
+                        if ($this.IsVfpRuleConfigured($endpoint.VfpPort, $allowNatRuleName, "SLB_GROUP_NAT_IPv4_OUT", "SLB_NAT_LAYER") -eq $false)
+                        {
+                            $this.Status = [TestStatus]::Failed
+                            $this.RootCause += ",Vfp port for POD with IP {0} has missing rule {1}" -f $endpoint.IPAddress, $ruleName
+                            $this.Resolution = "Restart HNS service by executing: Restart-Service -f HNS"
+                        }
+    
+                    }
+                }
+            }
+
+            if ($lbPolicy.ServiceType -eq [ServiceType]::NodePort) {
+                $ruleName = "LB_\w*_{0}_{1}_{2}_{3}" -f $networkData.ManagementIpAddress, $lbPolicy.ExternalPort, $lbPolicy.ExternalPort, $lbPolicy.Protocol
+            }
+            else {
+                $ruleName = "LB_\w*_{0}_{1}_{2}_{3}" -f $lbPolicy.VIP, $lbPolicy.ExternalPort, $lbPolicy.ExternalPort, $lbPolicy.Protocol
+            }
+
+            if ($this.IsVfpRuleConfigured($networkData.HostVfpPort, $ruleName, "SLB_GROUP_LB_IPv4_OUT", "SLB_LB_LAYER") -eq $false)
+            {
+                $this.Status = [TestStatus]::Failed
+                $this.RootCause += ",Vfp port for POD with IP {0} has missing rule {1}" -f $endpoint.IPAddress, $ruleName
```

Comment:
We are checking host port rules, not pod. Should this error message be for the host port, instead of particular pod?